frontstage/controllers/party_controller.py

```diff
@@ -229,32 +230,112 @@ def get_respondent_enrolments(party_id):
                 }
 
 
+class CachingDataThreadCreation(threading.Thread):
+    def __init__(self, function, *args):
+        threading.Thread.__init__(self)
+        self.function = function
+        self.args = args
+
+    def run(self):
+        self.function(*self.args)
+
+
+def set_enrolment_data(enrolment_data):
+    surveys_ids = set()
+    business_ids = set()
+    for enrolment in enrolment_data:
+        surveys_ids.add(enrolment['survey_id'])
+        business_ids.add(enrolment['business_id'])
+    return surveys_ids, business_ids
+
+
+def caching_data_for_survey_list(cache_data, surveys_ids, business_ids, tag):
+    # Creates a list of threads which will call functions to set the survey, case, party and collex responses
+    # in the cache_data.
+
+    threads = []
+
+    for survey_id in surveys_ids:
+        threads.append(CachingDataThreadCreation(get_survey, cache_data, app.config, survey_id))
+        threads.append(CachingDataThreadCreation(get_collex, cache_data, app.config, survey_id))
+
+    for business_id in business_ids:
+        threads.append(CachingDataThreadCreation(get_case, cache_data, app.config, business_id, tag))
+        threads.append(CachingDataThreadCreation(get_party, cache_data, app.config, business_id))
+
+    for thread in threads:
+        thread.start()
+    for thread in threads:
+        thread.join()
+
+
+def caching_data_for_collection_instrument(cache_data):
+    # This function creates a list of threads from the collection instrument id in the cache_data of the cases.
+    collection_instrument_ids = set()
+    threads = []
+    for business_id, cases in cache_data['cases'].items():
+        for case in cases:
+            collection_instrument_ids.add(case['collectionInstrumentId'])
+    for collection_instrument_id in collection_instrument_ids:
+        threads.append(CachingDataThreadCreation(get_collection_instrument, cache_data, app.config,
+                                                 collection_instrument_id))
+
+    for thread in threads:
+        thread.start()
+    for thread in threads:
+        thread.join()
+
+# This method has gone through a rewrite in an attempt to make the to-do page more performant. Before the rewrite, it
```

Comment:
When I read this I thought it was a comment about caching_data_for_collection_instrument. When its about get_survey_list_details_for_party . So I think it should be a docstring on get_survey_list_details_for_party ? I am not sure if the comment should reference old code , it should just say what the new code does

Reply:
Seems like a good idea to make it a docstring instead. Once this goes in everyone is gonna forget the old way of doing it.